mesonbuild/modules/gnome.py

```diff
@@ -536,7 +536,10 @@ def _scan_gir_targets(self, state, girtargets):
             if isinstance(girtarget, build.Executable):
                 ret += ['--program', girtarget]
             else:
-                libname = girtarget.get_basename()
+                if isinstance(girtarget, build.SharedLibrary):
+                    libname = girtarget.get_basename()
+                else:
+                    libname = os.path.join("@PRIVATE_OUTDIR_ABS_%s@" % girtarget.get_id(), girtarget.get_filename())
```

Comment:
Maybe add a comment here that it needs a g-i with https://gitlab.gnome.org/GNOME/gobject-introspection/merge_requests/72, and that's why we don't do it for shared libraries, otherwise it would break older g-i.

Reply:
Done.